dataflowengine/src/main/scala/io/shiftleft/dataflowengine/language/TrackingPoint.scala

```diff
@@ -32,10 +31,15 @@ class TrackingPoint(raw: GremlinScala[nodes.TrackingPoint]) extends NodeSteps[no
     )
   }
 
-  def reachableBy(sourceTravs: NodeSteps[nodes.TrackingPoint]*): TrackingPoint = {
+  /**
+    * Convert to nearest CFG node
+    * */
+  def cfgNode: Steps[nodes.CfgNode] = map(_.cfgNode)
+
+  def reachableBy[NodeType <: nodes.TrackingPoint](sourceTravs: Steps[NodeType]*): Steps[NodeType] = {
     val pathReachables = reachableByInternal(sourceTravs)
     val reachedSources = pathReachables.map(_.reachedSource)
-    new TrackingPoint(graph.asScala().inject(reachedSources: _*).asInstanceOf[GremlinScala[nodes.TrackingPoint]])
+    new NodeSteps[NodeType](graph.asScala().inject(reachedSources: _*).asInstanceOf[GremlinScala[NodeType]])
```

Comment:
haven't asked the compiler, but this should be equivalent: `__(reachedSources: _*)`

Reply:
Compiler agrees with you, fixed :)